src/app/pages/home/exchange/exchange.page.ts

```diff
@@ -152,11 +169,156 @@ export class ExchangePage implements OnInit {
           console.log(error);
         } else {
           console.log("Card: ", data)
+          this.isCardFound = true;
           this.card = data;
           this.frontImg = this.card.frontImg;
           this.backImg = this.card.backImg;
+          this.ownerId = this.authService.getCurrentUserId();
+          console.log('Current owner ID: ', this.ownerId);
+          this.holderId = this.card.holders;
+          console.log('Current holder ID: ', this.holderId);
         }
       });
     }
   }
+
+  async selectCard() {
+    const modal = await this.modalController.create({
+      component: CardSelectPage
+    });
+
+    modal.onWillDismiss().then(data => {
+       console.log(data);
+    });
+    return await modal.present();
+  }
+
+  cancel() {
+    !this.isCardFound;
+    this.router.navigate(['/home/exchange']);
+  }
+
+  startNFCListener() {
+
+    this.nfc.addNdefListener(() => {
+      console.log('successfully attached ndef listener');
+
+      let toast = this.toastCtrl.create({
+        message: 'Connected',
+        duration: 1000,
+        position: 'top'
+      }).then(toast => toast.present());
+
+    }, (err) => {
+      console.log('error attaching ndef listener', err);
+
+      let toast = this.toastCtrl.create({
```

Comment:
Is this needed twice? Lines 206 - 210 seem to be the same.

Reply:
If so, we could split this into its own function.

Reply:
I think so in order to indicate whether the ndefListener has been properly added or there was an error. I've never gotten the error toast message though. Splitting into a separate function could definitely help with building out the NFC functionality as clean as possible.